recon/src/recon/sqlitestore.rs

```diff
@@ -396,6 +413,31 @@ where
             .await?;
         Ok(row.map(|row| row.get(0)))
     }
+
+    #[instrument(skip(self))]
+    async fn keys_with_missing_values(
+        &mut self,
+        range: RangeOpen<Self::Key>,
+    ) -> Result<Vec<Self::Key>> {
+        let query = sqlx::query(
```

Comment:
when we need to check if the range is valid? In `hash_range`, we check if left >= right and return 0 but in other range related queries we don't.